src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -497,16 +489,17 @@ func (r *Cluster) validateRedpandaMemory() field.ErrorList {
 			field.Invalid(
 				field.NewPath("spec").Child("resources").Child("requests").Child("memory"),
 				r.Spec.Resources.Requests.Memory(),
-				"need 2GB of memory per core; need to decrease the requested CPU or increase the memory request"))
+				"requests.memory < 2Gi per core; either decrease requests.cpu or increase requests.memory"))
 	}
 
 	redpandaCores := r.Spec.Resources.RedpandaCPU().Value()
-	if !r.Spec.Resources.RedpandaMemory().IsZero() && r.Spec.Resources.RedpandaMemory().Value() < redpandaCores*MinimumMemoryPerCore {
+	minimumMemoryPerCore := int64(math.Floor(MinimumMemoryPerCore * RedpandaMemoryAllocationRatio))
```

Comment:
I'm not sure where the rule 2GB per core comes from, but if it comes from Redpanda core, shouldn't this be left unscaled? (although it may make existing clusters invalid)

Reply:
It's kind of arbitrary. So is 10% buffer. There are no (currently supported) cloud instance types that have <3.5GiB/core, so this is likely only for  test clusters or for people who are specifically tuning things, in which case they need to know what they're doing. Without `--overprovisioned` / `deverloperMode` Redpanda complains at <1GiB/core.

And yeah, as mentioned in the cover letter, this is for compatibility with existing configurations. Also, "2.223GiB / core is the minimum `requests.memory`" is awkward.